### PR TITLE
Feature/initial

### PR DESCRIPTION
**Pull request by sripberger**
I tried making some changes to our test environment with this one. Feel free to let me know if you have any questions about them.

### CODE REVIEW

**Comment by CrispyConductor** [comment]
Needs to be atomic.

File: README.md [hunk 1]
```diff
@@ -1,3 +1,87 @@
...
+let limiter = new RateLimiter(client);
+
+
+// At most one per 2 seconds.
+let check = () => limiter.check('key', 0.5, 1);
+
+// First check will resolve.
+check()
+	// Second check will reject.
+	.then(() => check().catch(() => {}))
```
Comment (rawsp33d) [inline README.md:30]: I get that this is just example code, but this line will always pass no matter how `check` is implemented.  You could add a `then` stanza in which you throw an error if you want, which would essentially flip the success bit.

Comment (rawsp33d) [reply README.md:30]: But it definitely conveys the point you are trying to convey :D Keep it!

**Comment by rosshadden** [comment]
LGTM.

**Review by rosshadden — APPROVED** [review]
LGTM.

File: README.md [hunk 2]
```diff
@@ -1,3 +1,87 @@
...
+```
+
+## Custom Prefix
+By default, `rediz-rate-limiter` prefixes keys with 'rzrate:' before storing
+them in redis. This prefix can be changed with the RateLimiter constructor.
+Keys with a different prefix will be separated from each other, even if they
+are otherwise the same.
+
+```js
+let asdfLimiter = new RateLimiter(client, 'asdf:');
```
Comment (CrispyConductor) [inline README.md]: I'd recommend always having a `rzrate:` prefix to namespace all rate limiter keys.  The second argument to the constructor should be appended to that.

Additionally, the second argument to the constructor shouldn't need to have a ':'.  That should also automatically be appended.

Comment (CrispyConductor) [reply README.md]: I'd also recommend allowing to specify a default rate and burst in the constructor, so if the last two arguments aren't given to the `check()` call, they use the defaults specified in the constructor.

File: lib/rate-limiter.js [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+	check(key, rate, burst) {
+		return this._scriptPromise
+			.then(() => {
+				let prefixedKey = this.prefix + key;
+				let shard = this.client.shard(key);
+				return shard.runScript(
+					'rateCheck',
+					`${prefixedKey}:count`,
+					`${prefixedKey}:timestamp`,
+					Date.now(),
```
Comment (CrispyConductor) [inline lib/rate-limiter.js:59]: Weren't you going to pass seconds into this?

**Comment by CrispyConductor** [comment]
Looks good, just a few minor things.

File: lib/rate-limiter.js (hunk 3 above)
Comment (sripberger) [reply lib/rate-limiter.js:59]: I considered getting the timestamp in lua, which would be in seconds. Opted to pass it in from js as per your original suggestion.

**Pull request merged**